table/src/main/java/org/enso/table/data/column/builder/PrimInferredStorageBuilder.java

```diff
@@ -0,0 +1,119 @@
+package org.enso.table.data.column.builder;
+
+import org.enso.table.data.column.DoubleStorage;
+import org.enso.table.data.column.LongStorage;
+import org.enso.table.data.column.Storage;
+
+import java.util.BitSet;
+
+/**
+ * A column builder for numeric types. Tries to interpret all data as 64-bit integers. If that
+ * becomes impossible, retypes itself to store 64-bit floats. When even that fails, falls back to a
+ * {@link StringStorageBuilder}.
+ */
+public class PrimInferredStorageBuilder extends StorageBuilder {
+  private enum Type {
+    LONG,
+    DOUBLE
```

Comment:
I really don't like how the storage types seem to be getting duplicated everywhere. Can we not create an enum that represents the storage types and use it everywhere?

Reply:
This enum is completely private for this builder. Will probably never grow. And things like `STRING` are straight invalid here. So this is actually the clean solution.